### PR TITLE
[FIX] website_sale: Coupon Code

### PR DESCRIPTION
**Pull request by simongoffin**
The pricelist linked to the coupon code is just applied in the cart.
In the shop, the pricelist of the user is applied.

opw:634639

### CODE REVIEW

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -934,11 +934,14 @@ def tracking_cart(self, **post):
         return ret
 
     @http.route(['/shop/get_unit_price'], type='json', auth="public", methods=['POST'], website=True)
-    def get_unit_price(self, product_ids, add_qty, **kw):
+    def get_unit_price(self, product_ids, add_qty, on_cart=False, **kw):
```
Comment (Gorash) [inline addons/website_sale/controllers/main.py]: Change attribute name like use_order_pricelist

File: addons/website_sale/controllers/main.py [hunk 2]
```diff
@@ -934,11 +934,14 @@ def tracking_cart(self, **post):
...
         cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
         products = pool['product.product'].browse(cr, uid, product_ids, context=context)
         partner = pool['res.users'].browse(cr, uid, uid, context=context).partner_id
-        pricelist_id = request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id
+        if on_cart:
+            pricelist_id = request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id
+        else:
+            pricelist_id = partner.property_product_pricelist.id
```
Comment (Gorash) [inline addons/website_sale/controllers/main.py]: pricelist_id = use_order_pricelist and request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id

**Comment by simongoffin** [comment]
@beledouxdenis

File: addons/website_sale/controllers/main.py (hunk 2 above)
Comment (Gorash) [inline addons/website_sale/controllers/main.py]: .

**Comment by jholze** [comment]
:+1:

**Pull request closed without merging**